core/internal/cltest/cltest.go

```diff
@@ -325,11 +325,23 @@ func NewApplication(t testing.TB, flagsAndDeps ...interface{}) *TestApplication
 
 // NewApplicationWithKey creates a new TestApplication along with a new config
 // It uses the native keystore and will load any keys that are in the database
-func NewApplicationWithKey(t *testing.T) *TestApplication {
+func NewApplicationWithKey(t *testing.T, flagsAndDeps ...interface{}) *TestApplication {
 	t.Helper()
 
 	config := NewTestGeneralConfig(t)
-	return NewApplicationWithConfigAndKey(t, config)
+	return NewApplicationWithConfigAndKey(t, config, flagsAndDeps...)
+}
+
+// NewApplicationWithKeyAndBalanceMonitorDisabled creates a new TestApplication along with a new config and the balance monitor disabled
+// It uses the native keystore and will load any keys that are in the database
+func NewApplicationWithKeyAndBalanceMonitorDisabled(t *testing.T, flagsAndDeps ...interface{}) *TestApplication {
```

Comment:
I don't think this is necessary, you can just use NewApplicationWithConfig and set the config override beforehand

Reply:
Resolved